src/test/java/com/zilen/weather/service/WeatherServiceTest.java

```diff
@@ -18,6 +18,10 @@
 
     @Mock
     private RestTemplate restTemplate;
+    @Value("${com.zilen.url}")
+    private String url;
+    @Value("${com.zilen.appid}")
+    private String appid;
```

Comment:
Why you inject these values in test? These variables belong to weather service, not to your test. Furthermore, Spring injection doesn't work here.